File: src/Python/538.BSTToGreaterTree.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    lastsum = 0
    def order(self, root):
        if root is None:
            return
        self.order(root.right)
        root.val += self.lastsum 
        self.lastsum = root.val
        self.order(root.left)

    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        self.order(root)
        return root

if __name__ == '__main__':
    solution = Solution()
    root = TreeNode(5)
    root.left = TreeNode(2)
    root.right = TreeNode(13)
    print(solution.convertBST(root))
    root = TreeNode(5)
    root.left = TreeNode(4)
    root.right = TreeNode(7)
    print(solution.convertBST(root))
else:
    pass

File: src/CSharp/538.BSTToGreaterTree.cs
using System;
using System.Collections.Generic;

namespace PeefyLeetCode.ConvertBST
{
    public class TreeNode
    {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
    }

public class Solution {
    int lastsum = 0;
    public TreeNode ConvertBST(TreeNode root) {
        if (root == null)
            return null;
        ConvertBST(root.right);
        root.val += lastsum;
        lastsum = root.val;
        ConvertBST(root.left);
        return root;
    }
}
}